regulations/generator/node_types.py

```diff
@@ -45,6 +49,22 @@ def label_to_text(label, include_section=True, include_marker=False):
             or _l2t_section(label, include_section, include_marker))
 
 
+_MARKERLESS_REGEX = re.compile(r'p\d+')
+
+
+def _paragraph_tail(label_parts, join_with, prefix='', suffix=''):
```

Comment:
I think that tests should focus on public functions; this private function is an implementation detail. I don't think it's necessary to establish via a test unless there's a known corner case.

Regarding what it's supposed to do, that's why there's a comment :)